post-processor.go

```diff
@@ -161,14 +163,57 @@ func (p *PostProcessor) PostProcess(ui packer.Ui, artifact packer.Artifact) (pac
 	}
 	if size > 100*1024*1024 {
 		ui.Message("File size > 100MB. Initiating multipart upload")
-		multi, err := p.s3.Multi(boxPath, "application/octet-stream", "public-read")
+
+		multi, err := p.s3.InitMulti(boxPath, "application/octet-stream", "public-read")
 		if err != nil {
 			return nil, false, err
 		}
-		parts, err := multi.PutAll(file, 5*1024*1024)
-		if err != nil {
-			return nil, false, err
+
+		ui.Message("Uploading...")
+
+		const chunkSize = 10*1024*1024
+
+		totalParts := uint64(math.Ceil(float64(size) / float64(chunkSize)))
+		totalUploadSize := int64(0)
+
+		parts := []s3.Part{}
+
+		errorCount := 0;
+
+		for partNum := uint64(1); partNum <= totalParts; partNum++ {
```

Comment:
It would be simpler to make this a loop over offsets rather than part numbers (like [`s3.PutAll`](https://github.com/mitchellh/goamz/blob/master/s3/multi.go#L290-L323))